modal/_container_entrypoint.py

```diff
@@ -846,6 +862,11 @@ def breakpoint_wrapper():
             int_handler = signal.signal(signal.SIGINT, signal.SIG_IGN)
             usr1_handler = signal.signal(signal.SIGUSR1, signal.SIG_IGN)
 
+            for finalized_function in finalized_functions.values():
+                if finalized_function.lifespan_manager:
+                    with container_io_manager.handle_user_exception():
+                        event_loop.run(finalized_function.lifespan_manager.lifespan_shutdown())
```

Comment:
Nice 👍

Just FYI this will run if there is an interrupt or shutdown between lines 845 and 849, which means that the session may not have been initialized. So on line 405 of `_asgi.py`, we may want to just silently ignore the missing session.

Reply:
makes sense! thanks!